erts/emulator/beam/erl_bif_lists.c

```diff
@@ -203,6 +233,23 @@ static BIF_RETTYPE lists_append_onheap(Process *p, ErtsAppendContext *context) {
 }
 
 static int append_continue(Process *p, ErtsAppendContext *context) {
+    /* Fast-lane when the rhs is nil: return lhs. */
+    if (is_nil(context->rhs_original)) {
+        int res;
+
+        /* First check that lhs is proper. */
+        res = append_is_proper_list(p, &context->iterator);
+
+        if (res == 1) {
+#ifdef DEBUG
+            context->result_cdr = &context->rhs_original;
+#endif
+            context->result = context->lhs_original;
+        }
+
+        return res;
```

Comment:
If we move setting these to the branch returning `1` in `append_is_proper_list`, we can shorten this a lot. While we're at it we can change its name to `append_empty_rhs` or the likes.

Otherwise this PR LGTM.